core/services/relay/delegate/delegate.go

```diff
@@ -0,0 +1,144 @@
+package delegate
+
+import (
+	"encoding/json"
+	"fmt"
+
+	solanaGo "github.com/gagliardetto/solana-go"
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana"
+	"github.com/smartcontractkit/chainlink/core/chains/evm"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/keystore"
+	"github.com/smartcontractkit/sqlx"
+
+	"github.com/pkg/errors"
+
+	uuid "github.com/satori/go.uuid"
+	"github.com/smartcontractkit/chainlink/core/services/job"
+	"github.com/smartcontractkit/chainlink/core/services/relay"
+	"github.com/smartcontractkit/chainlink/core/services/relay/ethereum"
+	"go.uber.org/multierr"
+)
+
+var (
+	_ relay.Relayer = &ethereum.Relayer{}
+	_ relay.Relayer = &solana.Relayer{}
+)
+
+type delegate struct {
+	relayers map[relay.Network]relay.Relayer
+	ks       keystore.Master
+}
+
+func NewRelayDelegate(db *sqlx.DB, ks keystore.Master, chainSet evm.ChainSet, lggr logger.Logger) *delegate {
+	return &delegate{
+		ks: ks,
+		relayers: map[relay.Network]relay.Relayer{
+			relay.Ethereum: ethereum.NewRelayer(db, chainSet, lggr),
+			relay.Solana:   solana.NewRelayer(lggr),
+		},
+	}
+}
+
+// A delegate relayer on start will start all relayers it manages.
+func (d delegate) Start() error {
+	var err error
+	for _, r := range d.relayers {
+		err = multierr.Combine(err, r.Start())
+	}
+	return err
+}
+
+// A delegate relayer on close will close all relayers it manages.
+func (d delegate) Close() error {
+	var err error
+	for _, r := range d.relayers {
+		err = multierr.Combine(err, r.Close())
+	}
+	return err
+}
+
+// A delegate relayer is healthy if all relayers it manages are ready.
+func (d delegate) Ready() error {
+	var err error
+	for _, r := range d.relayers {
+		err = multierr.Combine(err, r.Ready())
+	}
+	return err
+}
+
+// A delegate relayer is healthy if all relayers it manages are healthy.
+func (d delegate) Healthy() error {
+	var err error
+	for _, r := range d.relayers {
+		err = multierr.Combine(err, r.Healthy())
+	}
+	return err
+}
+
+func (d delegate) NewOCR2Provider(externalJobID uuid.UUID, s interface{}) (relay.OCR2Provider, error) {
+	spec, ok := s.(*job.OffchainReporting2OracleSpec)
+	if !ok {
+		return nil, errors.New("unsuccessful cast to 'job.OffchainReporting2OracleSpec'")
```

Comment:
IMO its reasonable to panic here since it can't come from untrusted input and would indicate a programming error